app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsImageRecsDialog.kt

```diff
@@ -0,0 +1,90 @@
+package org.wikipedia.suggestededits
+
+import android.app.Dialog
+import android.os.Bundle
+import android.widget.CompoundButton
+import androidx.appcompat.app.AlertDialog
+import androidx.core.os.bundleOf
+import androidx.fragment.app.DialogFragment
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import org.wikipedia.R
+import org.wikipedia.activity.FragmentUtil
+import org.wikipedia.databinding.FragmentSuggestedEditsImageRecsDialogBinding
+
+class SuggestedEditsImageRecsDialog : DialogFragment() {
+    interface Callback {
+        fun onDialogSubmit(response: Int, selectedItems: List<Int>)
+    }
+
+    private var _binding: FragmentSuggestedEditsImageRecsDialogBinding? = null
+    private val binding get() = _binding!!
+    private var dialog: AlertDialog? = null
+
+    private var responseCode = -1
+
+    private val checkBoxChangedHandler = CompoundButton.OnCheckedChangeListener { _, _ ->
+        updateSubmitState()
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
+    }
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        responseCode = requireArguments().getInt(ARG_RESPONSE)
+        _binding = FragmentSuggestedEditsImageRecsDialogBinding.inflate(layoutInflater)
+
+        binding.checkBox1.setOnCheckedChangeListener(checkBoxChangedHandler)
+        binding.checkBox2.setOnCheckedChangeListener(checkBoxChangedHandler)
+        binding.checkBox3.setOnCheckedChangeListener(checkBoxChangedHandler)
+        binding.checkBox4.setOnCheckedChangeListener(checkBoxChangedHandler)
+        binding.checkBox5.setOnCheckedChangeListener(checkBoxChangedHandler)
+        binding.checkBox6.setOnCheckedChangeListener(checkBoxChangedHandler)
```

Comment:
Would it be possible to put all the `checkbox` views in an ArrayList, and use that ArrayList to apply listener, add item, and check enable state?